### DISCUSSION TITLE
MomentJS increases 80KB+ First Load JS

### DISCUSSION THREAD

**Discussion by phibersoft** [vercel/next.js#23044] [Help]
### Story

I began to investigate why it was so slow while writing my new project. 
I reviewed all the big packages by building them separately. I finally came to the conclusion.

### Build Debugs

// Basic App - Cloned from nextjs github

Page                                                           Size     First Load JS
┌ ○ /                                                          3.46 kB        66.5 kB
├   └ css/c50ddf22b716c7b6b76d.css                             660 B
├   /_app                                                      0 B              63 kB
├ ○ /404                                                       3.46 kB        66.5 kB
└ λ /api/hello                                                 0 B              63 kB
+ First Load JS shared by all                                  63 kB
  ├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js  13.4 kB
  ├ chunks/framework.e2fe4a.js                                 41.8 kB
  ├ chunks/main.f37b48.js                                      6.64 kB
  ├ chunks/pages/_app.a2d915.js                                529 B
  ├ chunks/webpack.50bee0.js                                   751 B
  └ css/6e9ef204d6fd7ac61493.css                               194 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
   (ISR)     incremental static regeneration (uses revalidate in getStaticProps)

// After Redux Integration

Page                                                           Size     First Load JS
┌ ○ /                                                          3.46 kB        74.9 kB
├   └ css/c50ddf22b716c7b6b76d.css                             660 B
├   /_app                                                      0 B            71.5 kB
├ ○ /404                                                       3.46 kB        74.9 kB
└ λ /api/hello                                                 0 B            71.5 kB
+ First Load JS shared by all                                  71.5 kB
  ├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js  13.4 kB
  ├ chunks/framework.b1290c.js                                 42.1 kB
  ├ chunks/main.f37b48.js                                      6.64 kB
  ├ chunks/pages/_app.0ad3d3.js                                8.63 kB
  ├ chunks/webpack.50bee0.js                                   751 B
  └ css/a747e6f315ccdc249aab.css                               399 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
   (ISR)     incremental static regeneration (uses revalidate in getStaticProps)

// After Material UI integration

Page                                                           Size     First Load JS
┌ ○ /                                                          3.46 kB        92.9 kB
├   └ css/c50ddf22b716c7b6b76d.css                             660 B
├   /_app                                                      0 B            89.5 kB
├ ○ /404                                                       3.46 kB        92.9 kB
└ λ /api/hello                                                 0 B            89.5 kB
+ First Load JS shared by all                                  89.5 kB
  ├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js  13.4 kB
  ├ chunks/framework.b1290c.js                                 42.1 kB
  ├ chunks/main.f37b48.js                                      6.64 kB
  ├ chunks/pages/_app.e01445.js                                26.6 kB
  ├ chunks/webpack.50bee0.js                                   751 B
  └ css/a747e6f315ccdc249aab.css                               399 B

// After "only copy" Components folder -- i didn't use any component in this folder

Page                                                           Size     First Load JS
┌ ○ /                                                          3.46 kB         179 kB
├   └ css/c50ddf22b716c7b6b76d.css                             660 B
├   /_app                                                      0 B             175 kB
├ ○ /404                                                       3.46 kB         179 kB
└ λ /api/hello                                                 0 B             175 kB
+ First Load JS shared by all                                  175 kB
  ├ chunks/75fc9c18.b43118.js                                  19.2 kB
  ├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js  13.4 kB
  ├ chunks/framework.b1290c.js                                 42.1 kB
  ├ chunks/main.021315.js                                      6.64 kB
  ├ chunks/pages/_app.62557c.js                                93.2 kB
  ├ chunks/webpack.50bee0.js                                   751 B
  └ css/a747e6f315ccdc249aab.css                               399 B

// After delete the component includings "moment.js"

Page                                                           Size     First Load JS
┌ ○ /                                                          3.46 kB        93.1 kB
├   └ css/c50ddf22b716c7b6b76d.css                             660 B
├   /_app                                                      0 B            89.6 kB
├ ○ /404                                                       3.46 kB        93.1 kB
├ λ /api/hello                                                 0 B            89.6 kB
└ λ /profile/[id]                                              325 B          89.9 kB
+ First Load JS shared by all                                  89.6 kB
  ├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js  13.4 kB
  ├ chunks/framework.b1290c.js                                 42.1 kB
  ├ chunks/main.f37b48.js                                      6.64 kB
  ├ chunks/pages/_app.ad98b7.js                                26.8 kB
  ├ chunks/webpack.50bee0.js                                   751 B
  └ css/a747e6f315ccdc249aab.css                               399 B

### Question

If the problem is that it just includes "momentjs" completely, then I have no other way to delete the package. Isn't it a mistake what I don't understand? For example, suppose I'm just using "moment (). Unix ()". In this case, it includes the whole package.

**Comment by masterkingmk** [comment]
Hello dude.i am very weak in programming and I surprised why all of people in world want to give up me?i found I don’t need any crypto coin new bueilding now because all of assesses is don’t watching to me all of my trade income is going’s?if you want help me help to end this and I see my income and assists not with a code send me.for one time I create this and all of this is get down I don’t do one work two time.i found a lot site is being to me but I never see income more than 11 years.i see 11 years transaction but I don’t know where are money.

Sent from Yahoo Mail for iPad


On Sunday, March 14, 2021, 05:53, Adem ***@***.***> wrote:




Story

I began to investigate why it was so slow while writing my new project.
I reviewed all the big packages by building them separately. I finally came to the conclusion.

Build Debugs

// Basic App - Cloned from nextjs github

Page Size First Load JS
┌ ○ / 3.46 kB 66.5 kB
├ └ css/c50ddf22b716c7b6b76d.css 660 B
├ /_app 0 B 63 kB
├ ○ /404 3.46 kB 66.5 kB
└ λ /api/hello 0 B 63 kB
   
   - First Load JS shared by all 63 kB   
├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js 13.4 kB   
├ chunks/framework.e2fe4a.js 41.8 kB   
├ chunks/main.f37b48.js 6.64 kB   
├ chunks/pages/_app.a2d915.js 529 B   
├ chunks/webpack.50bee0.js 751 B   
└ css/6e9ef204d6fd7ac61493.css 194 B

λ (Server) server-side renders at runtime (uses getInitialProps or getServerSideProps)
○ (Static) automatically rendered as static HTML (uses no initial props)
● (SSG) automatically generated as static HTML + JSON (uses getStaticProps)
(ISR) incremental static regeneration (uses revalidate in getStaticProps)

// After Redux Integration

Page Size First Load JS
┌ ○ / 3.46 kB 74.9 kB
├ └ css/c50ddf22b716c7b6b76d.css 660 B
├ /_app 0 B 71.5 kB
├ ○ /404 3.46 kB 74.9 kB
└ λ /api/hello 0 B 71.5 kB
   
   - First Load JS shared by all 71.5 kB   
├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js 13.4 kB   
├ chunks/framework.b1290c.js 42.1 kB   
├ chunks/main.f37b48.js 6.64 kB   
├ chunks/pages/_app.0ad3d3.js 8.63 kB   
├ chunks/webpack.50bee0.js 751 B   
└ css/a747e6f315ccdc249aab.css 399 B

λ (Server) server-side renders at runtime (uses getInitialProps or getServerSideProps)
○ (Static) automatically rendered as static HTML (uses no initial props)
● (SSG) automatically generated as static HTML + JSON (uses getStaticProps)
(ISR) incremental static regeneration (uses revalidate in getStaticProps)

// After Material UI integration

Page Size First Load JS
┌ ○ / 3.46 kB 92.9 kB
├ └ css/c50ddf22b716c7b6b76d.css 660 B
├ /_app 0 B 89.5 kB
├ ○ /404 3.46 kB 92.9 kB
└ λ /api/hello 0 B 89.5 kB
   
   - First Load JS shared by all 89.5 kB   
├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js 13.4 kB   
├ chunks/framework.b1290c.js 42.1 kB   
├ chunks/main.f37b48.js 6.64 kB   
├ chunks/pages/_app.e01445.js 26.6 kB   
├ chunks/webpack.50bee0.js 751 B   
└ css/a747e6f315ccdc249aab.css 399 B

// After "only copy" Components folder -- i didn't use any component in this folder

Page Size First Load JS
┌ ○ / 3.46 kB 179 kB
├ └ css/c50ddf22b716c7b6b76d.css 660 B
├ /_app 0 B 175 kB
├ ○ /404 3.46 kB 179 kB
└ λ /api/hello 0 B 175 kB
   
   - First Load JS shared by all 175 kB   
├ chunks/75fc9c18.b43118.js 19.2 kB   
├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js 13.4 kB   
├ chunks/framework.b1290c.js 42.1 kB   
├ chunks/main.021315.js 6.64 kB   
├ chunks/pages/_app.62557c.js 93.2 kB   
├ chunks/webpack.50bee0.js 751 B   
└ css/a747e6f315ccdc249aab.css 399 B

// After delete the component includings "moment.js"

Page Size First Load JS
┌ ○ / 3.46 kB 93.1 kB
├ └ css/c50ddf22b716c7b6b76d.css 660 B
├ /_app 0 B 89.6 kB
├ ○ /404 3.46 kB 93.1 kB
├ λ /api/hello 0 B 89.6 kB
└ λ /profile/[id] 325 B 89.9 kB
   
   - First Load JS shared by all 89.6 kB   
├ chunks/f6078781a05fe1bcb0902d23dbbb2662c8d200b3.9d55af.js 13.4 kB   
├ chunks/framework.b1290c.js 42.1 kB   
├ chunks/main.f37b48.js 6.64 kB   
├ chunks/pages/_app.ad98b7.js 26.8 kB   
├ chunks/webpack.50bee0.js 751 B   
└ css/a747e6f315ccdc249aab.css 399 B

Question

If the problem is that it just includes "momentjs" completely, then I have no other way to delete the package. Isn't it a mistake what I don't understand? For example, suppose I'm just using "moment (). Unix ()". In this case, it includes the whole package.

**Comment by metin1** [comment]
I don't use moment Js at my new project. I used only js native date and time functions. I need only the date packages at the material ui date picker. I used dayjs instead of the moment.js there.  Dayjs is only 2.8kb (minified) (https://bundlephobia.com/result?p=dayjs@1.10.4). Why do you need moment js at your project?
